### ISSUE TITLE
Alternative for Cosine Similarity

### ISSUE THREAD

**Issue by vdpappu** [etherlabsio/ai-engine#82]
Currently, we use Cosine Similarity for similarity metric. With complex architectures like BERT, it may not be effective as the objective functions used for pre-training or fine-tuning does not directly reflect sentence relatedness without labelled dataset. Siamese Network (https://www.youtube.com/watch?v=6jfw8MuKwpI) provides an effective alternative for similarity tasks. This involves training a small network whose outputs highlights the similarity/dissimilarity between two inputs.
